L1Trigger/L1THGCalUtilities/plugins/ntuples/HGCalTriggerNtupleHGCMulticlusters.cc

```diff
@@ -115,20 +134,34 @@ void HGCalTriggerNtupleHGCMulticlusters::fill(const edm::Event& e, const edm::Ev
     cl3d_srrtot_.emplace_back(cl3d_itr->sigmaRRTot());
     cl3d_srrmax_.emplace_back(cl3d_itr->sigmaRRMax());
     cl3d_srrmean_.emplace_back(cl3d_itr->sigmaRRMean());
-    cl3d_emaxe_.emplace_back(cl3d_itr->eMax() / cl3d_itr->energy());
+    cl3d_emaxe_.emplace_back(cl3d_itr->eMax()/cl3d_itr->energy());
     cl3d_bdteg_.emplace_back(id_->value(*cl3d_itr));
     cl3d_quality_.emplace_back(cl3d_itr->hwQual());
 
+    //Per layer cluster information
+    int layer;
+    std::vector<float> layer_pt(52,0.0);
+    for(const auto cl_ptr : cl3d_itr->constituents())
```

Comment:
The copy is probably not very expensive (pair<int,Ptr>), but it's probably better to use a reference `const auto&`.